sdks/python/apache_beam/transforms/ptransform.py

```diff
@@ -1080,8 +1080,14 @@ def callable_ptransform_factory(*args, **kwargs):
 
   # The signature of this PTransform constructor is that of fn minus the first
   # argument (which is where the pvalue is passed during expand).
-  callable_ptransform_factory.__signature__ = inspect.signature(  # type: ignore
-      functools.partial(fn, None))
+  try:
+    inspect.signature(fn)
+  except Exception:
+    # If we can't get a signature, don't attempt to patch it up.
+    pass
+  else:
+    callable_ptransform_factory.__signature__ = inspect.signature(  # type: ignore
+        functools.partial(fn, None))
```

Comment:
Any reason not to just stick the whole assignment call in the `try`? I'm guessing there aren't cases where the functools call changes the effectiveness of grabbing the signature, but it would be a bit safer (as an additional minor plus, it saves us the extra inspection)

Reply:
I was thinking that if anything else goes wrong (it shouldn't) we'd want to know (and generally try to keep catchng generic Exception especially as small as possible). But I could go either way if you have a preference.

Reply:
I have a soft preference towards catching more because (a) this likely only matters if things _do_ break, and (b) it things do break then this is a breaking change across versions, which I'd prefer to avoid.

Reply:
OK. Done.